climada/hazard/centroids/centr.py

```diff
@@ -820,386 +638,274 @@ def get_closest_point(self, x_lon, y_lat, scheduler=None):
         idx_close : int
             Index of centroid in internal ordering of centroids.
         """
-        if self.meta:
-            if not self.lat.size or not self.lon.size:
-                self.set_meta_to_lat_lon()
-            i_lat, i_lon = rasterio.transform.rowcol(self.meta['transform'], x_lon, y_lat)
-            i_lat = np.clip(i_lat, 0, self.meta['height'] - 1)
-            i_lon = np.clip(i_lon, 0, self.meta['width'] - 1)
-            close_idx = int(i_lat * self.meta['width'] + i_lon)
-        else:
-            self.set_geometry_points(scheduler)
-            close_idx = self.geometry.distance(Point(x_lon, y_lat)).values.argmin()
+        close_idx = self.geometry.distance(Point(x_lon, y_lat)).values.argmin()
         return self.lon[close_idx], self.lat[close_idx], close_idx
 
-    def set_region_id(self, scheduler=None):
-        """Set region_id as country ISO numeric code attribute for every pixel or point.
+    # NOT REALLY AN ELEVATION FUNCTION, JUST READ RASTER
+    def get_elevation(self, topo_path):
+        """Return elevation attribute for every pixel or point in meters.
 
         Parameters
         ----------
-        scheduler : str
-            used for dask map_partitions. “threads”, “synchronous” or “processes”
+        topo_path : str
+            Path to a raster file containing gridded elevation data.
+
+        Returns
+        -------
+        values : np.array of shape (npoints,)
+            Interpolated elevation values from raster file for each given coordinate point.
         """
-        ne_geom = self._ne_crs_geom(scheduler)
-        LOGGER.debug('Setting region_id %s points.', str(self.lat.size))
-        self.region_id = u_coord.get_country_code(
-            ne_geom.geometry[:].y.values, ne_geom.geometry[:].x.values)
+        return u_coord.read_raster_sample(topo_path, self.lat, self.lon)
 
-    def set_area_pixel(self, min_resol=1.0e-8, scheduler=None):
-        """Set `area_pixel` attribute for every pixel or point (area in m*m).
+    def get_dist_coast(self, signed=False, precomputed=False):
+        """Get dist_coast attribute for every pixel or point in meters.
```

Comment:
We have to be careful when merging this because https://github.com/CLIMADA-project/climada_python/pull/840 is also making changes here.